lib/puma/launcher.rb

```diff
@@ -521,5 +523,21 @@ def with_unbundled_env
         Bundler.with_unbundled_env { yield }
       end
     end
+
+    def log_config
+      log "Configuration:"
+
+      @config.options.default_options
```

Comment:
Hmmm. In practice, I'd rather that this method just called something like `@config.final_options` and then iterate through those. Then, we also use _that_ method when calculating the final config. That method may not exist yet, you may have to refactor it out and use it in both places.

Normally I wouldn't be so eager to DRY up 2 places, but in this case it's critical for correctness.

Reply:
OK, I see now why that doesn't exist, because this would be the first place in Puma we try to iterate through every option.

So I think we need a new method on UserFileDefaultOptions that basically does what you do here.

Reply:
I agree.  This issue has been on my list, as our use of a Server object in CI (and also by capybara) means that the Server object needs to have a bunch of default values in it's initialize method, which is a PITA and prone to error, as they exist in `Puma::Configuration#puma_default_options`...